### PR TITLE
Stop replaying truncated Responses streams (Fixes #3049)

### LINKED ISSUE #3049: OpenAI Responses HTTP/SSE: mid-stream retry concatenates attempts and abrupt EOF is treated as success
## Summary

Two defects in the OpenAI Responses HTTP/SSE path, found while fixing #3034.

## 1. The HTTP stream retry can concatenate attempts

`fetchStreamWithRetries` in
`packages/providers/src/openai-responses/openAIResponsesHttpStream.ts` wraps
`yield* parseSuccessfulResponse(...)` in its attempt-level `try`/`catch`. If the
body fails mid-stream *after* messages were already yielded, it retries the
whole request and yields the second attempt's messages into the SAME provider
iterator. Downstream, `StreamProcessor` accumulates the whole iterator into one
assistant message, so the partial and the regenerated response are concatenated.

This is exactly what `RetryOrchestrator` forbids one layer up
(`yieldStreamUnprotected` -> `markErrorAfterStreamOutput`, "cannot retry (would
produce mixed response)"). The WebSocket transport deliberately does NOT do this
(see #3034 / PR #3047); the HTTP path should match.

Additionally, the same `retries` ephemeral drives both this internal loop and the
orchestrator's budget, and the provider does not declare
`transportAttemptOwnership = 'provider'`, so pre-output failures can multiply
into roughly attempts x attempts requests.

### Suggested fix

Track whether any `IContent` has been yielded for the current attempt and only
retry when nothing has been emitted; otherwise rethrow. Then reconcile the
attempt budget so the internal loop and the orchestrator do not multiply.

## 2. An abrupt SSE EOF is accepted as a successful response

`parseResponsesStream`
(`packages/providers/src/openai/parseResponsesStream.ts`) returns normally when
the reader reaches EOF, without requiring a terminal event, and
`parseSuccessfulResponse` adds no completion check. A connection cut mid-body
therefore looks like a completed (but truncated) response. Agent-side validation
(`packages/agents/src/core/streamValidationHelpers.ts`) only rejects a missing
finish reason when there is also no text, so a truncated answer is silently
committed to history.

The WebSocket transport already asserts that a terminal response event was
received. The HTTP path should apply the equivalent check and raise a
stream-interruption error instead of silently truncating.

## Acceptance criteria

1. An HTTP stream that fails after partial output is not internally replayed.
2. Internal and orchestrator attempts do not multiply.
3. An SSE body that ends without a terminal response event raises a
   stream-interruption error rather than completing successfully.
4. Behavioral tests for both, following the existing Responses executor test
   style.

Follow-up to #3034 / PR #3047.

### PR DESCRIPTION
**Pull request by acoliver**
## TLDR

Fixes the OpenAI Responses HTTP/SSE path so it cannot concatenate output from multiple transport attempts and cannot silently accept a truncated stream.

The transport now stops retrying once an attempt has yielded content, consumes internal retries from the same transport-attempt budget as RetryOrchestrator, and requires an accepted terminal Responses event before treating SSE EOF as successful.

## Dive Deeper

Three related failure modes are addressed:

1. A body error after a text delta previously caused the HTTP transport to replay the entire request inside the same async iterator. Downstream code then accumulated the partial first response and regenerated second response into one assistant message. The HTTP retry loop now records whether an attempt yielded any IContent and rethrows immediately after output instead of replaying.

2. The internal HTTP retry loop and RetryOrchestrator previously treated the retries setting as independent budgets, allowing physical fetch attempts to multiply. Internal retries after the first attempt now consume from the attached shared TransportAttemptBudget. Direct executor use without an orchestrator retains the existing maxStreamingAttempts behavior.

3. Reader EOF previously completed successfully even when the SSE body never contained response.completed, response.done, or response.incomplete. The HTTP path now opts into terminal-event enforcement and raises StreamInterruptionError for abrupt EOF. WebSocket behavior remains unchanged; both transports share the accepted terminal-event definitions.

Terminal recognition is monotonic per dispatched event rather than inferred from logging state, so an accepted terminal followed by another valid event behaves identically regardless of reader chunk boundaries. Later response.failed and error events still retain their specific error precedence.

Existing HTTP fixtures that model successful Responses streams now include a real terminal event rather than relying on the data-done sentinel alone.

## Reviewer Test Plan

1. Run the behavioral regression file:

       bun test packages/providers/src/openai-responses/openAIResponsesExecutor.streamIntegrity.test.ts

   It covers:
   - no replay after partial output;
   - legitimate retry before output;
   - shared attempt-budget enforcement;
   - abrupt EOF rejection;
   - successful terminal completion;
   - all accepted terminal event types followed by a nonterminal event;
   - reader chunk-boundary independence;
   - protocol-failure precedence.

2. Run the providers test workspace and verify all provider tests pass.

3. Run typecheck, lint, format, and build.

4. Optionally use a local OpenAI-compatible SSE endpoint that closes after a delta without sending a terminal response event. Verify the request fails as a stream interruption and is not silently committed as a completed assistant response.

Local verification:

- New behavioral tests: 10/10 passed.
- Providers package: 501/501 test files passed.
- Typecheck: passed.
- Build: passed.
- Format: passed.
- ESLint policy guard and focused ESLint: passed.
- Full lint passed during implementation verification; a second foreground invocation was externally terminated before producing output.
- Smoke test with the stepfun-37 profile: passed.
- Full npm test run reached one unrelated core test hook timeout under parallel load in config.mcp-lazy.test.ts; that test passed 9/9 immediately in isolation. No provider tests failed.
- DeepThinker review found one terminal-tracking edge case; it was remediated with chunk-boundary regression coverage.
- Open Code Review reviewed 25 changed files with zero findings. Its review of the large new behavioral test exhausted the tool-round budget; that file was covered by DeepThinker and the focused 10-test run.

## Testing Matrix

|          | 🍏  | 🪟  | 🐧  |
| -------- | --- | --- | --- |
| npm run  | ✅  | ❓  | ❓  |
| npx      | ❓  | ❓  | ❓  |
| Docker   | ❓  | ❓  | ❓  |
| Podman   | ❓  | -   | -   |
| Seatbelt | ❓  | -   | -   |

## Linked issues / bugs

Fixes #3049

Follow-up to #3034 and PR #3047.

### CODE REVIEW

File: packages/providers/src/openai-responses/openAIResponsesExecutor.streamIntegrity.test.ts [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+        }),
+    );
+    const single = await drainWithPossibleRejection(
+      executeOpenAIResponsesRequest(buildNormalizedOptions(), buildDeps()),
+    );
+
+    // The identical two events split across separate reader chunks.
+    fetchMock = installFetch(
+      async () => new Response(encodeSse([fixture.event, CREATED_EVENT])),
+    );
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/providers/src/openai-responses/openAIResponsesExecutor.streamIntegrity.test.ts:430]: Fixed in commit 420fceb94c. The test now restores the first fetch mock and clears the local reference before installing the split-chunk mock, so afterEach always restores the real fetch implementation. The behavioral file passes 10/10 tests, and focused ESLint and Prettier checks pass.

File: packages/providers/src/openai-responses/openAIResponsesHttpStream.ts [hunk 2]
```diff
@@ -142,14 +144,34 @@ async function* fetchStreamWithRetries(
...
+    // count as the orchestrator, not retries × retries. When no budget is
+    // attached (direct unit tests without an orchestrator),
+    // tryConsumeTransportAttempt returns true and today's
+    // maxStreamingAttempts-only behavior is unchanged.
+    if (
+      streamingAttempt > 1 &&
+      !tryConsumeTransportAttempt(params.normalizedOptions)
+    ) {
+      throw lastError;
+    }
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/providers/src/openai-responses/openAIResponsesHttpStream.ts:162]: Evaluated against transportAttemptBudget.ts, retryTransportOwnership.ts, and RetryOrchestrator.ts. This concern is not valid. attachTransportAttemptBudget stores one mutable TransportAttemptBudget object in request metadata, and every getTransportAttemptBudget or hasTransportAttemptRemaining call reads that same object and its current used property. No caller caches a used snapshot for retry eligibility. RetryOrchestrator does capture usedBefore only to prevent double-accounting in accountProviderAttempt; after an internal retry increments the shared object, the inequality deliberately suppresses a second increment. The AC2 behavioral test exercises the real orchestrator and executor and observes exactly N physical fetches for a budget of N. No production change is needed.
